### PR TITLE
feat(engine): sort items topologically

### PR DESCRIPTION
**Pull request by W95Psp**
- [x] add tests

### CODE REVIEW

**Review by karthikbhargavan — APPROVED** [review]
Much needed. Do we need to test this on all our code to be sure?

File: engine/lib/dependencies.ml [hunk 1]
```diff
@@ -205,6 +206,15 @@ module Make (F : Features.T) = struct
   let ident_list_to_string =
     List.map ~f:Concrete_ident.DefaultViewAPI.show >> String.concat ~sep:", "
 
+  let sort (items : item list) : item list =
+    let g = ItemGraph.of_items items in
+    let lookup (name : concrete_ident) =
+      List.find ~f:(ident_of >> Concrete_ident.equal name) items
+    in
+    ItemGraph.Topological.fold List.cons g []
+    (* |> List.rev *)
```
Comment (karthikbhargavan) [inline engine/lib/dependencies.ml:215]: delete?

Comment (W95Psp) [reply engine/lib/dependencies.ml:215]: Oh, right, thanks!

**Comment by W95Psp** [comment]
I added an assert to make sure the reordering wasn't losing any item, turned out "orphan" items were discarded! I fixed that now.
So I thing merging this is fine.

**Comment by W95Psp** [comment]
Ok, the CI is happy, I'm pressing merge

**Pull request merged**